address.go

```diff
@@ -26,19 +26,18 @@ type Address struct {
 }
 
 type AddressRequest struct {
-	XMLName            xml.Name `xml:"address"`
```

Comment:
Had to remove `XMLName` as I got 
`payment_method_integration_test.go:48: xml: name "billing-address" in tag of braintree.PaymentMethodRequest.BillingAddress conflicts with name "address" in *braintree.AddressRequest.XMLName`
This should be fine. But I have some problems running all integrations test against my sandbox. @leighmcculloch can you verify this? Otherwise I would need to create a new type `BillingAddressRequest` with another `XMLName`

Reply:
I'll verify this.

Reply:
So removing this makes this work for cases where `AddressRequest` is used as a sub-element where it's top-level XMLName comes from the parent, but `AddressRequest` is also used as a root XML node in the `AddressGateway` and this causes it to serialize with the root tag name `AddressRequest` instead of `address` which is not accepted by the API.

Reply:
I think we can make `AddressRequest` work for both if we keep the `XMLName` field and set it's `Local` field to be the name we need in the `AddressGateway` case. e.g.:

```patch
From caa5b7554902b05c347d8fb1f24effc3c3374625 Mon Sep 17 00:00:00 2001
From: Leigh McCulloch <leigh@mcchouse.com>
Date: Wed, 17 Apr 2019 05:26:15 +0000
Subject: Fix use of AddressRequest in AddressGateway

---
 address.go         | 1 +
 address_gateway.go | 2 ++
 2 files changed, 3 insertions(+)

diff --git a/address.go b/address.go
index 9e9d3ee..6f2060d 100644
--- a/address.go
+++ b/address.go
@@ -26,6 +26,7 @@ type Address struct {
 }
 
 type AddressRequest struct {
+       XMLName            xml.Name
        FirstName          string `xml:"first-name,omitempty"`
        LastName           string `xml:"last-name,omitempty"`
        Company            string `xml:"company,omitempty"`
diff --git a/address_gateway.go b/address_gateway.go
index e0b7f73..8db254b 100644
--- a/address_gateway.go
+++ b/address_gateway.go
@@ -10,6 +10,7 @@ type AddressGateway struct {
 
 // Create creates a new address for the specified customer id.
 func (g *AddressGateway) Create(ctx context.Context, customerID string, a *AddressRequest) (*Address, error) {
+       a.XMLName.Local = "address"
        resp, err := g.execute(ctx, "POST", "customers/"+customerID+"/addresses", &a)
        if err != nil {
                return nil, err
@@ -36,6 +37,7 @@ func (g *AddressGateway) Delete(ctx context.Context, customerId, addrId string)
 
 // Update updates an address for the address id and customer id.
 func (g *AddressGateway) Update(ctx context.Context, customerID, addrID string, a *AddressRequest) (*Address, error) {
+       a.XMLName.Local = "address"
        resp, err := g.execute(ctx, "PUT", "customers/"+customerID+"/addresses/"+addrID, a)
        if err != nil {
                return nil, err
-- 
2.20.1
```

Reply:
Tried this. But the `Test_grpcServer_GetTransaction` test fails with:
`expected tx2.RiskData not to be empty`
 
@leighmcculloch, do you have any idea why this happens?